hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -371,27 +350,50 @@ public void deleteByControllerID(final String controllerID) {
     }
 
     @Override
-    public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetID) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetID);
+    public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetId) {
+        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
 
-        return targetRepository.findByAssignedDistributionSetId(pageReq, distributionSetID);
+        return targetRepository.findByAssignedDistributionSetId(pageReq, distributionSetId);
     }
 
     @Override
-    public Page<Target> findByAssignedDistributionSetAndRsql(final Pageable pageReq, final long distributionSetID,
-            final String rsqlParam) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetID);
+    public Page<Target> findByAssignedDistributionSetAndQuery(final Pageable pageable, final long distributionSetId,
+            final String query) {
+        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
+        // Maybe optimize in order to fetch only the IDs from the database:
+        List<String> controllerIds = targetRepository
+                .findByAssignedDistributionSetId(Pageable.unpaged(), distributionSetId).stream()
+                .map(t -> t.getControllerId()).collect(Collectors.toList());
+        return convertPage(targetQueryExecutionManagement.findByQuery(pageable, query, controllerIds), pageable);
+    }
+
+    // Used by Auto-Assign checker: Identify new targets which have not an
```

Comment:
fix format of this comments, and maybe convert into a standard javadoc format?

Reply:
Its is a override method. So converting to standard javadoc will override the javadoc from the interface method. Do something like below:
`
 /*
     * (non-Javadoc) Used by Auto-Assign checker: Identify new targets which
     * have not an specific DS assigned.
     */
`

Reply:
The comment was not intended to occure in the java-doc but was a hint for developers looking in the code.